publisher-sdk/src/main/java/com/criteo/publisher/privacy/ConsentData.kt

```diff
@@ -16,28 +16,22 @@
 
 package com.criteo.publisher.privacy
 
+import android.content.SharedPreferences
 import com.criteo.publisher.annotation.OpenForTesting
-import java.util.concurrent.atomic.AtomicReference
 
 @OpenForTesting
-class ConsentData {
-
-  /**
-   * Consent can be in any of the following two states:
-   *
-   * <ul>
-   *  <li> CONSENT_GIVEN: it is known that consent has been given by the user</li>
-   *  <li> CONSENT_NOT_GIVEN: it is known that consent has not been given by the user,
-   *  or isn't known yet.</li>
-   * </ul>
-   */
-  enum class ConsentStatus {
-    CONSENT_GIVEN,
-    CONSENT_NOT_GIVEN
+class ConsentData(val sharedPreferences: SharedPreferences) {
+  companion object {
+    const val CRITEO_CONSENT_GIVEN_KEY = "CRTO_ConsentGiven"
   }
-  private val valueRef = AtomicReference<ConsentStatus>(ConsentStatus.CONSENT_NOT_GIVEN)
 
-  var consentStatus: ConsentStatus
-    get() = valueRef.get()
-    set(value) = valueRef.set(value)
+  @Synchronized
```

Comment:
Why do you need those methods to be synchronized ?

Reply:
read & writes happening on different threads & unsynchronized code.

Reply:
From the `SharedPreference` javadoc:

> Note: This class provides strong consistency guarantees. It is using expensive operations which might slow down an app. Frequently changing properties or properties where loss can be tolerated should use other mechanisms. For more details read the comments on SharedPreferences.Editor.commit() and SharedPreferences.Editor.apply().

And for `SharedPreferences.Editor.apply()`:

> This atomically performs the requested modifications, replacing whatever is currently in the SharedPreferences.

Both reads are consistent and writes are atomic. So you don't need to synchronize the access even when used from different threads.

Reply:
Yes, but if I want to read the latest available value then it is helpful to synchronize (since writing involves grabbing the `editor` reference first before calling `apply`). Though I agree it's not mandatory. I can remove.